### ISSUE TITLE
Array constructor with params

### ISSUE THREAD

**Issue by dorba** [nikhilk/scriptsharp#60]
Would it be possible to add a constructor to the Array object, taking (params object[ ]) ? It seems reasonable since the javascript Array object can handle this natively. And if so, it would also be great to also have available: new List<T>(params T[ ]).

E.g, something like:
List<string> vals = new List<string>("one", "two"); 

Compiled to:
Array vals = new Array("one", "two");

**Comment by dorba** [comment]
Eh, it removed the generics from list, but I'm sure you get the idea.

**Comment by robitar** [comment]
@Barry,

The core libs are up there now, you might take a look at https://github.com/nikhilk/scriptsharp/blob/master/src/Libraries/CoreLib/Collections/Generic/List.cs and see if you can hack it in.

You should just be able to add a constructor overload accepting a params T[], as list is already mapped to a native Array.

**Comment by nikhilk** [comment]
Here is a problem with this one...

If one declares a List of int, then is new List(3) creating an array with capacity == 3 or is it creating an array with a single item, 3, in it.

**Comment by robitar** [comment]
Its a list containing 3 elements, which is what would happen if you did the same in native script. That's an edge case that Visual Studio can help you with, it clearly binds the single int arg to the int constructor. If you want to add a single int to the list, then just use the default constructor and explicitly call add. I think the advantage of having the params T[] syntax would be worth this minor ambiguity.

``` c#
namespace ScriptLibrary {
    class MyClass {
        void F() {
            List<int> a = new List<int>(5); //binds to ctor(int) in intellisense
            List<int> b = new List<int>(5, 6); //binds to ctor(T[])
        }
    }
}
```

**Comment by nikhilk** [comment]
As per the discussion, the fix to add ArrayList and List<T> ctor taking a set of initial values has been added, and included in the 0.7.3 release.
